elastic/kibana
# [Snapshot Restore] Disable navigation when a step is invalid

## Summary

Fix https://github.com/elastic/kibana/issues/95876

The fix implemented here was to prevent users from navigating to a different step if the current step is invalid. This mirrors the behaviour in the es_ui_shared `FormWizard` (which should ideally replace all wizards in the future).

## How to test

Assuming you have no snapshots to restore:

1. Start ES & Kibana and navigate to Snapshot & Restore (NB; you should start ES setting `-E path.repo=/tmp/es-snaps`)
2. In Snapshot and restore set up a new repository
3. Set up a new policy
4. Run the policy
5. Go to the snapshots list and select one to restore (should only have one policy)
6. Test that the behaviour in the restore wizard is addressed

## Release note

We fixed an issue where users can get stuck in the snapshot restore wizard if they navigate "Back" on a step that contains errors because the "next" button would be disabled.
